utilities/scripts/xmlPagsToEntityPags.groovy

```diff
@@ -0,0 +1,199 @@
+    def entityFile = new File(entityDir, filename)
+    println "Creating file $entityFile.absolutePath for XML PAGS group $groupName"
+
+    def writer = new BufferedWriter(new FileWriter(entityFile))
+    def builder = new MarkupBuilder(writer)
+    builder.mkp.xmlDeclaration(version: '1.0', encoding: 'UTF-8');
+    builder.'pags-group'(script: 'classpath://org/jasig/portal/io/import-pags-group_v4-1.crn') {
+        name(groupName)
+        description(group.'group-description'.text())
+        // If there is a members element, create members but use the group-name instead of the group-key.
+        group.members.each { membersItem ->
+            members() {
+                membersItem.'member-key'.each() {
+                    if (pagsGroupNames.get(it.text()) != null) {
+                        'member-name'(pagsGroupNames.get(it.text()))
+                    } else {
+                        println "WARN PAGS group ${groupName} has member-key ${it.text()} which does not map to" +
+                                " a group in PAGS.  Ignoring invalid member"
+                    }
+                }
+            }
+        }
+        // If there is a selection test, create a selection-test node
+        group.'selection-test'.each { selectionTest ->
+            'selection-test'() {
+                selectionTest.'test-group'.each { testGroup ->
+                    'test-group'() {
+                        testGroup.test.each { testItem ->
+                            test() {
+                                // Handle mobile user agent string of ${mobile.user.agent.regex}. Change tester from
+                                // RegexTester to PropertyRegexTester and InvertedRegexTester to PropertyInvertedRegexTester
+                                // and use the property string for the property testers.
+                                if (testItem.'test-value'.text().contains("{mobile.user.agent.regex}")) {
+                                    'attribute-name'(testItem.'attribute-name'.text())
+                                    'tester-class'(testItem.'tester-class'.text()
+                                            == "org.jasig.portal.groups.pags.testers.RegexTester" ?
+                                            "org.jasig.portal.groups.pags.testers.PropertyRegexTester" :
+                                            "org.jasig.portal.groups.pags.testers.PropertyInvertedRegexTester")
+                                    'test-value'("org.jasig.portal.http.header.userAgent.mobile.regex.pattern")
+                                } else {
+                                    'attribute-name'(testItem.'attribute-name'.text())
+                                    'tester-class'(testItem.'tester-class'.text())
+                                    'test-value'(testItem.'test-value'.text())
+                                }                                          pagsXmlToEntity
```

Comment:
Argh.  Thanks for catching.  I didn't see that.  Slow IDE.  Occassionally I do a search or something and focus doesn't always switch to the find dialog before I've typed in the characters and they show up in the file instead.